src/twisted/trial/newsfragments/12390.bugfix

```diff
@@ -0,0 +1 @@
+Functions added via ``TestCase.addCleanup`` now are executed taking test case timeout into account.
```

Comment:
A minor sugestion to get the release notes with the canonical name of the targeted API 

```suggestion
twisted.trial.unittest.TestCase.addCleanup will cause the test to fail if the returned deferred is not resolved before the test's timeout.
```

Reply:
Fixed, thanks.